service/history/task/transfer_active_task_executor.go

```diff
@@ -198,8 +202,10 @@ func (t *transferActiveTaskExecutor) processActivityTask(
 	// the rest of logic is making RPC call, which takes time.
 	release(nil)
 
-	// Ratelimiting is not done. This is only to count the number of requests via metrics
-	t.wfIDCache.AllowInternal(task.DomainID, task.WorkflowID)
+	// Rate limiting task processing requests
```

Comment:
What is the reason to apply this for activity and decision tasks only? Are you planning to expand to [other task types](https://github.com/uber/cadence/blob/38930faa283559aca2232975f6d2493e99681d5c/service/history/task/transfer_active_task_executor.go#L130) later on?

Reply:
A lot of processing boils down to these two task types and they are easier to reason about when we ratelimit users. But it is possible to expand if and when needed

Reply:
Child workflow related tasks are also common and cause hot shards. I recommend including them in the scope